lib/ui/recommendation.js

```diff
@@ -150,7 +154,7 @@ Recommendation.prototype = {
           this.el.appendChild(recommendation);
         }
         this.el.collapsed = false;
-        this.data = data;
+        this.data = payload;
```

Comment:
If we just don't null out `this.data` on line 163 and 174, we should be able to get rid of `this.prev`, I think?